### PR TITLE
X axis grouping options now available.

### PR DESCRIPTION
**Pull request by rob-duplock**

### CODE REVIEW

File: src/Plugin/Visualisation/Style/AxisChart.php [hunk 1]
```diff
@@ -128,6 +128,7 @@ public function defaultConfiguration() {
    */
   public function settingsForm(array $form, FormStateInterface $form_state) {
     $form = parent::settingsForm($form, $form_state);
+    $dvf_helpers = \Drupal::service('dvf.helpers');
```
Comment (rob3000) [inline src/Plugin/Visualisation/Style/AxisChart.php]: @rob-duplock dependency injection.

File: src/Plugin/Visualisation/Style/AxisChart.php [hunk 2]
```diff
@@ -662,6 +690,38 @@ public function removeGridLine(array &$form, FormStateInterface $form_state) {
...
+   * @param array $form
+   *   The form structure.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   The updated form element.
+   */
+  public function updateColumnOverrides(array $form, FormStateInterface $form_state) {
+    $form_state->setRebuild(TRUE);
```
Comment (rob3000) [inline src/Plugin/Visualisation/Style/AxisChart.php]: Just curious but why do you need to trigger a rebuild?

**Review by rob3000 — CHANGES_REQUESTED** [review]

File: src/Plugin/Visualisation/Style/AxisChart.php (hunk 1 above)
Comment (rob-duplock) [reply src/Plugin/Visualisation/Style/AxisChart.php]: All instances of `\Drupal::service('dvf.helpers')` have been replaced with dependency injection

File: src/Plugin/Visualisation/Style/AxisChart.php (hunk 2 above)
Comment (rob-duplock) [reply src/Plugin/Visualisation/Style/AxisChart.php]: For whatever reason I thought these was required before returning the updated form element. Anyway, it works fine without it so I have removed the two instances of `rebuild()`

**Review by rob3000 — APPROVED** [review]

**Pull request merged**